### PR TITLE
Get extra_vars and stdout from Job

### PR DESCRIPTION
**Pull request by bzwei**

### CODE REVIEW

File: spec/job_spec.rb [hunk 1]
```diff
@@ -16,4 +18,37 @@
...
+      it "returns a hashed value" do
+        obj = described_class.new(instance_double("AnsibleTowerClient::Api"), raw_instance)
+        expect(obj.extra_vars_hash).to eq('option' => 'lots of options')
+      end
+    end
+
+    describe "#extra_vars does not exist" do
+      it "returns an empty hash" do
+        obj = described_class.new(instance_double("AnsibleTowerClient::Api"), raw_instance_no_extra_vars)
+        expect(obj.extra_vars_hash).to be_a_kind_of(Hash)
```
Comment (bdunne) [inline spec/job_spec.rb]: You could `expect(obj.extra_vars_hash).to eq({})` to reduce the number of expectations.

File: spec/factories/responses.rb [hunk 2]
```diff
@@ -15,17 +15,18 @@
     type       { AnsibleTowerClient::FactoryHelper.underscore_string(klass.namespace.last) }
     name       { "#{type}-#{id}" }
     url        { "/api/v1/#{klass.endpoint}/#{id}/" }
-    related    { {"survey_spec" => "example.com/api", 'inventory' => 'inventory link'} }
+    related    { {"survey_spec" => "example.com/api", 'inventory' => 'inventory link', 'stdout' => 'example.com/api'} }
```
Comment (bdunne) [inline spec/factories/responses.rb]: Do we want to have 'stdout' on every object?

File: lib/ansible_tower_client/job.rb [hunk 3]
```diff
@@ -3,6 +3,15 @@ class Job < BaseModel
...
       "jobs".freeze
     end
+
+    def extra_vars_hash
+      extra_vars.empty? ? {} : hashify(:extra_vars)
+    end
+
+    def stdout
+      out_url = related['stdout']
+      return nil unless out_url
```
Comment (bdunne) [inline lib/ansible_tower_client/job.rb]: The `nil` is not required.

**Comment by bdunne** [comment]
@bzwei Please rebase.

File: spec/factories/responses.rb (hunk 2 above)
Comment (bzwei) [inline spec/factories/responses.rb]: We don't have too, but `stdout` presents in every job's related section.

**Pull request merged**